modules/exadsBidAdapter.js

```diff
@@ -0,0 +1,465 @@
+  const gdprConsent = getGdprConsentChoice(bidderRequest);
+  const envParams = getEnvParams();
+
+  // Make a dynamic bid request to the ad partner's endpoint
+  let bidRequestData = {
+    'id': validBidRequest.bidId, // NOT bid.bidderRequestId or bid.auctionId
+    'at': 1,
+    'imp': [],
+    'site': {
+      'id': validBidRequest.params.siteId,
+      'domain': envParams.domain,
+      'cat': validBidRequest.params.catIab,
+      'page': envParams.page,
+      'keywords': validBidRequest.params.keywords
+    },
+    'device': {
+      'ua': envParams.userAgent,
+      'ip': validBidRequest.params.userIp,
+      'geo': {
+        'country': validBidRequest.params.country
+      },
+      'language': envParams.lang,
+      'os': envParams.osName,
+      'js': 0,
+      'ext': {
+        'accept_language': envParams.language
+      }
+    },
+    'user': {
+      'id': validBidRequest.params.userId,
+    },
+    'ext': {
+      'sub': 0,
+      'prebid': {
+        'channel': {
+          'name': 'pbjs',
+          'version': '$prebid.version$'
+        }
+      }
+    }
+  };
+
+  if (gdprConsent && gdprConsent.gdprApplies) {
+    bidRequestData.user['ext'] = {
+      consent: gdprConsent.consentString
+    }
+  }
+
+  // Banner setup
+  const bannerMediaType = utils.deepAccess(validBidRequest, 'mediaTypes.banner');
+  if (bannerMediaType != null) {
+    bidRequestData.imp = bannerMediaType.sizes.map(size => {
+      let ext;
+
+      if (validBidRequest.params.image_output || validBidRequest.params.video_output) {
+        ext = {
+          image_output: validBidRequest.params.image_output ? validBidRequest.params.image_output : undefined,
+          video_output: validBidRequest.params.video_output ? validBidRequest.params.video_output : undefined,
+        }
+      }
+
+      return ({
+        'id': validBidRequest.params.impressionId,
+        'bidfloor': validBidRequest.params.bidfloor,
+        'bidfloorcur': validBidRequest.params.bidfloorcur,
+        'banner': {
+          'w': size[0],
+          'h': size[1],
+          'mimes': validBidRequest.params.mimes ? validBidRequest.params.mimes : undefined,
+          ext
+        },
+      });
+    });
+  }
+
+  const nativeMediaType = utils.deepAccess(validBidRequest, 'mediaTypes.native');
+
+  if (nativeMediaType != null) {
+    const nativeVersion = '1.2';
+
+    const native = {
+      'native': {
+        'ver': nativeVersion,
+        'plcmttype': 4,
+        'plcmtcnt': validBidRequest.params.native.plcmtcnt
+      }
+    };
+
+    native.native.assets = bidRequestData.imp = nativeMediaType.ortb.assets.map(asset => {
+      const newAsset = asset;
+      if (newAsset.img != null) {
+        newAsset.img.wmin = newAsset.img.h;
+        newAsset.img.hmin = newAsset.img.w;
+      }
+      return newAsset;
+    });
+
+    const imp = [{
+      'id': validBidRequest.params.impressionId,
+      'bidfloor': validBidRequest.params.bidfloor,
```

Comment:
What if bidfloor is not set, is that removed from the `imp` object?

Reply:
It would be better if it was removed from the request if it's empty or 0, but I should work anyway so I guess it's fine